### PR TITLE
grpc: fix the wrong error handler.

### LINKED ISSUE #5373: PD will panic  if  tikv enable forwarding
## Bug Report
image

### What did you do?
enable forwarding  for tikv

### What did you expect to see?
nomal
### What did you see instead?
panic
### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?

Issue Number: Close #5373

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

### Release note

```release-note
grpc: fix the wrong error handler 
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: tests/client/client_test.go [hunk 1]
```diff
@@ -723,23 +728,44 @@ func TestClientTestSuite(t *testing.T) {
...
-	suite.srv, suite.cleanup, err = server.NewTestServer(assertutil.CheckerWithNilAssert(re))
-	suite.NoError(err)
-	suite.grpcPDClient = testutil.MustNewGrpcClient(re, suite.srv.GetAddr())
-	suite.grpcSvr = &server.GrpcServer{Server: suite.srv}
+	suite.ctx, suite.clean = context.WithCancel(context.Background())
+	svrs, cleanup := newMultiServer(suite.ctx, assertutil.CheckerWithNilAssert(re))
+	suite.cleanup = cleanup
+	suite.srv = server.MustWaitLeader(re, svrs)
+
+	//follower := ""
```
Comment (nolouch) [inline tests/client/client_test.go:736]: format

Comment (bufferflies) [reply tests/client/client_test.go]: fix

**Review by nolouch — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Comment by rleungx** [comment]
The test has race problem.

**Pull request merged**